transformer_engine/common/utils.py

```diff
@@ -6,7 +6,7 @@
 import warnings
 from enum import Enum
 
-warnings.simplefilter('default')
+warnings.filterwarnings("module", category=DeprecationWarning, module="utils")
```

Comment:
We're not matching the right module:
```suggestion
warnings.filterwarnings(
    "default",
    category=DeprecationWarning,
    module="transformer_engine.common.utils",
)
```
With this fix, I see the expected behavior with the following test case:
```python
import warnings
warnings.simplefilter("error")
import transformer_engine.jax as te
print(te.MajorShardingType.SINGLE)  # Print warning
print(te.MajorShardingType.DP)  # No warning
x = te.ShardingResource()  # Print warning
warnings.warn("hi", DeprecationWarning)  # Error
```
One thing I don't understand is the difference between `action="default"` and `action="module"` since I see the same behavior with both. [The docs](https://docs.python.org/3/library/warnings.html#the-warnings-filter) say that `action="module"` only prints the first warning in a module, but that doesn't seem to be happening.